src/tiled/command.cpp

```diff
@@ -50,6 +51,34 @@ QString Command::finalCommand() const
     return finalCommand;
 }
 
+QString Command::finalWorkingDir() const
+{
+    QString finalWorkingDir;
+
+    MapDocument *mapDoc = DocumentManager::instance()->currentDocument();
+    if (mapDoc) {
+        const QString mapPath = mapDoc->path();
+        finalWorkingDir.replace(QLatin1String("%executablepath"),
+                                QLatin1String("\"%1\"")).arg(executablePath());
+        finalWorkingDir.replace(QLatin1String("%mappath"),
+                                QLatin1String("\"%1\"")).arg(mapPath);
+
+    }
+    return finalWorkingDir;
+}
+
+QString Command::executablePath() const
+{
+    QString finalcommand = finalCommand();
+
+    // removing parameters from command string
+    QString executableFilename(finalcommand.mid(0, finalcommand.lastIndexOf(QLatin1String(" ")) - 1));
```

Comment:
The beginning of the parameters is assumed to be the last space character? That would only cover the single-parameter case. I'm not sure how this can be done properly... maybe executable and parameters should be separated already in the UI?